kratos/tests/cpp_tests/geometries/test_line_2d_2.cpp

```diff
@@ -19,42 +20,310 @@
+    * Tests the Jacobian determinants using 'GI_GAUSS_4' integration method.
+    */
+    KRATOS_TEST_CASE_IN_SUITE(Line2D2DeterminantOfJacobianIndex4, KratosCoreGeometriesFastSuite) {
+        auto geom = GeneratePointsUnitXDirectionLine2D2();
+        double JacobianDeterminant = 0.0;
+        const double ExpectedJacobian = 0.5;
+
+        JacobianDeterminant = geom->DeterminantOfJacobian( 1, GeometryData::GI_GAUSS_4 );
+        KRATOS_CHECK_NEAR(JacobianDeterminant, ExpectedJacobian, TOLERANCE);
+
+        JacobianDeterminant = geom->DeterminantOfJacobian( 2, GeometryData::GI_GAUSS_4 );
+        KRATOS_CHECK_NEAR(JacobianDeterminant, ExpectedJacobian, TOLERANCE);
+
+        JacobianDeterminant = geom->DeterminantOfJacobian( 3, GeometryData::GI_GAUSS_4 );
+        KRATOS_CHECK_NEAR(JacobianDeterminant, ExpectedJacobian, TOLERANCE);
+
+        JacobianDeterminant = geom->DeterminantOfJacobian( 4, GeometryData::GI_GAUSS_4 );
+        KRATOS_CHECK_NEAR(JacobianDeterminant, ExpectedJacobian, TOLERANCE);
+    }
+
+    /** Tests the Jacobian determinants using 'GI_GAUSS_4' integration method.
+    * Tests the Jacobian determinants using 'GI_GAUSS_4' integration method.
+    */
+    KRATOS_TEST_CASE_IN_SUITE(Line2D2DeterminantOfJacobianIndex5, KratosCoreGeometriesFastSuite) {
+        auto geom = GeneratePointsUnitXDirectionLine2D2();
+        double JacobianDeterminant = 0.0;
+        const double ExpectedJacobian = 0.5;
+
+        JacobianDeterminant = geom->DeterminantOfJacobian( 1, GeometryData::GI_GAUSS_5 );
+        KRATOS_CHECK_NEAR(JacobianDeterminant, ExpectedJacobian, TOLERANCE);
+
+        JacobianDeterminant = geom->DeterminantOfJacobian( 2, GeometryData::GI_GAUSS_5 );
+        KRATOS_CHECK_NEAR(JacobianDeterminant, ExpectedJacobian, TOLERANCE);
+
+        JacobianDeterminant = geom->DeterminantOfJacobian( 3, GeometryData::GI_GAUSS_5 );
+        KRATOS_CHECK_NEAR(JacobianDeterminant, ExpectedJacobian, TOLERANCE);
+
+        JacobianDeterminant = geom->DeterminantOfJacobian( 4, GeometryData::GI_GAUSS_5 );
+        KRATOS_CHECK_NEAR(JacobianDeterminant, ExpectedJacobian, TOLERANCE);
+
+        JacobianDeterminant = geom->DeterminantOfJacobian( 5, GeometryData::GI_GAUSS_5 );
+        KRATOS_CHECK_NEAR(JacobianDeterminant, ExpectedJacobian, TOLERANCE);
+    }
+
+    KRATOS_TEST_CASE_IN_SUITE(Line2D2ShapeFunctionsValues, KratosCoreGeometriesFastSuite) {
+        auto geom = GeneratePointsDiagonalLine2D2();
+        array_1d<double, 3> coord(3);
+        coord[0] = 2.0/3.0;
+        coord[1] = 2.0/3.0;
+        coord[2] = 0.0;
+        KRATOS_CHECK_NEAR(geom->ShapeFunctionValue(0, coord), 1.0/6.0, TOLERANCE);
+        KRATOS_CHECK_NEAR(geom->ShapeFunctionValue(1, coord), 5.0/6.0, TOLERANCE);
+        auto& r_geom = *geom;
+        auto p_geom_nodes = Line2D2<Node<3>>::Pointer(new Line2D2<Node<3>>(
+        Node<3>::Pointer(new Node<3>(r_geom[0].X(), r_geom[0].Y(), r_geom[0].Z())),
```

Comment:
what is the Id of such a node? it should not compile...

Reply:
True, probably is doing the conversion of the first value to an intener or considering 0 the id, I will look at the constructor

Reply:
In all, constructors id is mandatory:

https://github.com/KratosMultiphysics/Kratos/blob/babb118499d88a9cfe8b4ec5b460c34057715ec4/kratos/includes/node.h#L197

Maybe we should add an explicit to avoid these problems